### ISSUE TITLE
docs: Update all docs related to Nydus

### ISSUE THREAD

**Issue by zvonkok** [kata-containers/kata-containers#10138]
The current documentation about Nydus and for each of the use-cases, Kata, Confidential Containers, encrypted, guest pull is scattered around and confusing. 
We need to consolidate all the documentation and create an e2e document describing all the use cases and configurations with Nydus. 
We should target this for the next Kata release. 

Examples include: https://github.com/confidential-containers/guest-components/blob/main/image-rs/docs/pull_image_with_nydus.md 
https://github.com/kata-containers/kata-containers/blob/main/docs/design/kata-guest-image-management-design.md

@fidencio @gkurz @stevenhorsman

**Comment by stevenhorsman** [comment]
Yeah, I guess that trying to reconcile the kata-containers documentation with the CoCo documentation is the challenge as both communities probably don't want to soley rely on and link to the other, but keeping them consistent and organised as possible still makes sense.
